server/src/dbAdapter.ts

```diff
@@ -0,0 +1,36 @@
+import { Dynamo } from './dynamodb';
+import { Disk } from './diskdb';
+import { Result, DatabaseType, Database } from '../../frontend/src/types';
+
+export class DatabaseAdapter implements Database {
+  public type: DatabaseType;
+  db: Database;
+
+  constructor(type: DatabaseType) {
+    this.type = type;
+
+    if (this.type === DatabaseType.Dynamo) {
```

Comment:
You can just do this in `app.ts`, as Darren suggests you don't really need the adaptor